orion/controllers/achieve/achieveController.go

```diff
@@ -0,0 +1,108 @@
+package achieve
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+func GetAchievements(c *gin.Context) {
+	// Query Repo
+	achieveList := GetAllAchievements()
+
+	// JSON Response
+	c.JSON(http.StatusOK, achieveList)
+	return
+}
+
+func GetAchievement(c *gin.Context) {
+	// Incoming parameters
+	id64, _ := strconv.ParseUint(c.Param("id"), 10, 32)
+	id := uint(id64)
+
+	// Query Repo
+	achieve, err := GetAchievementById(id)
+	if err != nil {
+		panic(err)
+	} else {
+		c.JSON(http.StatusOK, achieve)
+	}
+	return
+}
+
+func CreateAchievement(c *gin.Context) {
+	// Incoming JSON
+	var achieveJson Achieve
+	c.BindJSON(&achieveJson)
+
+	if err := CheckValidAchievement(achieveJson); err != nil {
+		c.String(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Query Repo (INSERT & SELECT)
+	err := InsertAchievement(achieveJson)
+	if err != nil {
+		panic(err)
+	} else {
+		c.JSON(http.StatusOK, nil)
+	}
+	return
+}
+
+func UpdateAchievement(c *gin.Context) {
+	// Incoming JSON & Parameters
+	id64, _ := strconv.ParseUint(c.Param("id"), 10, 32)
+	id := uint(id64)
+	var achieveJson Achieve
+	c.BindJSON(&achieveJson)
+
+	if err := CheckValidAchievement(achieveJson); err != nil {
+		c.String(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	// Query Repo (UPDATE & SELECT)
+	err := UpdateAchievementById(id, achieveJson)
+	if err != nil {
+		panic(err)
+	} else {
+		c.JSON(http.StatusOK, nil)
+	}
+	return
+}
+
+func DeleteAchievement(c *gin.Context) {
+	// Incoming Parameters
+	id64, _ := strconv.ParseUint(c.Param("id"), 10, 32)
+	id := uint(id64)
+
+	// Query Repo (DELETE)
+	err := DeleteAchievementById(id)
+	if err != nil {
+		panic(err)
+	} else {
+		c.String(http.StatusOK, "Deleted Achieve " + string(id))
+	}
+	return
+}
+
+func CheckValidAchievement(achieve Achieve) error {
+	// Retrieves the inputted values
+	year := achieve.Year
+	message := achieve.Message
+
+	// Year validation
+	if year < 2000 {
+		return errors.New("invalid year")
```

Comment:
Let's make error messages all start with a Capital letter for just the first word.
`Invalid year`

Reply:
The go specifications say that error strings should not be capitalized.

Reply:
The Go specifications say that error strings should be lowercase @ahsu1230

Reply:
oh You're right. nvm
https://github.com/golang/go/wiki/Errors